src/3d/Main.fst

```diff
@@ -201,15 +256,25 @@ let process_file (en:env) (fn:string) (modul:string) (emit_fstar:bool) (emit_out
   : ML env =
   
   let _decls, t_decls, static_asserts, en =
-    translate_module en modul fn in
-  if emit_fstar then emit_fstar_code en modul t_decls static_asserts emit_output_types_defs
+      translate_module en modul fn
+  in
+  if emit_fstar 
+  then (
+    if Options.get_interpret()
+    then emit_fstar_code_for_interpreter en modul t_decls
+    else emit_fstar_code en modul t_decls;
+    emit_entrypoint en modul t_decls static_asserts emit_output_types_defs
+  )
   else IO.print_string (Printf.sprintf "Not emitting F* code for %s\n" fn);
 
   let ds = Binding.get_exported_decls en.binding_env modul in
   
   { binding_env = Binding.finish_module en.binding_env modul ds;
     typesizes_env = TypeSizes.finish_module en.typesizes_env modul ds;
-    translate_env = Translate.finish_module en.translate_env modul ds }
+    translate_env = 
+      if Options.get_interpret()
+      then en.translate_env
+      else Inl (Translate.finish_module (left en.translate_env) modul ds) }

```

Comment:
This seems to make `--interpret` break with modules (#41) in non-batch mode. More precisely, in `src/3d/tests/modules/src`, running `3d.exe Test.3d`, `3d.exe --batch Test.3d` or `3d.exe --batch --interpret Test.3d` all succeed (and in the non-batch case only produces .fst files for Test, as expected, by loading its .3d dependencies without producing F* code for those), but running `3d.exe --interpret Test.3d` fails with:
```
Fatal error: exception (Failure "Type decl not found for Point._TWO_D_POINT")
```
Without `--interpret`, when a single X.3d file is passed to 3d, its dependencies are loaded and their contents are added to the environment (in the `else` branch) but .fst files are generated only for X.
With `--interpret`, the environment does not seem to be changed, or is changed only if `.fst` files are actually generated, which is not the case for dependencies. Is my understanding correct? If so, then why do we need to strictly tie populating the environment with generating F* code?

Reply:
I think this comes from that line in `InterpreterTarget.translate_decls`:
https://github.com/project-everest/everparse/blob/617f21ebf635c83da457619d1225a110a21f5ef1/src/3d/InterpreterTarget.fst#L480
Is there any way to disentangle the call to `translate_decls` from `Main.emit_fstar_code_from_interpreter`?

Reply:
thanks Tahina. I fixed this